diff --git a/packages/isl/package.desc b/packages/isl/package.desc
index ffae478d..20f7e024 100644
--- a/packages/isl/package.desc
+++ b/packages/isl/package.desc
@@ -1,6 +1,6 @@
 repository='git git://repo.or.cz/isl.git'
 bootstrap='./autogen.sh'
-mirrors='http://isl.gforge.inria.fr'
+mirrors='https://downloads.sourceforge.net/project/libisl/'
 relevantpattern='*.*|.'
 milestones='0.12 0.13 0.14 0.15 0.18'
 archive_formats='.tar.xz .tar.bz2 .tar.gz'
