diff --git a/packages/expat/package.desc b/packages/expat/package.desc
index 4e3e1a01..84dba8ff 100644
--- a/packages/expat/package.desc
+++ b/packages/expat/package.desc
@@ -1,6 +1,6 @@
 repository='git https://github.com/libexpat/libexpat.git'
 repository_subdir='expat'
 bootstrap='./buildconf.sh && make -C doc all'
-mirrors='http://downloads.sourceforge.net/project/expat/expat/${CT_EXPAT_VERSION}'
-archive_formats='.tar.bz2'
+mirrors='http://downloads.sourceforge.net/project/expat/expat/${CT_EXPAT_VERSION} https://github.com/libexpat/libexpat/releases/download/R_${CT_EXPAT_VERSION//./_}'
+archive_formats='.tar.xz .tar.lz .tar.bz2 .tar.gz'
 relevantpattern='*.*|.'
